openedx/core/lib/api/view_utils.py

```diff
@@ -104,16 +104,8 @@ def _wrapper(self, request, *args, **kwargs):
                         depth=depth,
                         check_if_enrolled=True,
                     )
-                except Http404:
-                    # any_unfulfilled_milestones called a second time since has_access returns a bool
-                    if check_for_milestones and any_unfulfilled_milestones(course_id, request.user.id):
-                        message = {
-                            "developer_message": "Cannot access content with unfulfilled "
-                                                 "pre-requisites or unpassed entrance exam."
-                        }
-                        return response.Response(data=message, status=status.HTTP_204_NO_CONTENT)
-                    else:
-                        raise
+                except Http404 as error:
+                    return response.Response(data=error.message, status=status.HTTP_204_NO_CONTENT)
```

Comment:
Note that according to the RFC, "The 204 response MUST NOT include a message-body".